dandi/dandiapi.py

```diff
@@ -1378,6 +1391,8 @@ class BaseRemoteAsset(ABC, APIBase):
     identifier: str = Field(alias="asset_id")
     #: The asset's (forward-slash-separated) path
     path: str
+    #: The path within the asset, as e.g. path in a zarr/
+    subpath: str | None = Field(default=None)
```

Comment:
I don't like the idea of a `BaseRemoteAsset` object pointing to a resource inside an asset; the class is supposed to be for assets only.  It'd be better to introduce a separate class for Zarr prefixes (and possibly also add a variant of `get_asset_by_path()` for returning this separate class).

Reply:
Yeah, I also disliked it a little but it is already quite a hierarchy of classes there leading to nearly combinatorial explosion.  If you see a sensible way, e.g. through adding a single extra class - please go ahead.